pilot/pkg/bootstrap/server.go

```diff
@@ -971,14 +985,31 @@ func (s *Server) initIstiodCerts(args *PilotArgs, host string) error {
 		}
 	} else if features.EnableCAServer && features.PilotCertProvider == constants.CertProviderIstiod {
 		log.Infof("initializing Istiod DNS certificates host: %s, custom host: %s", host, features.IstiodServiceCustomHost)
-		err = s.initDNSCerts()
+		err = s.initDNSCertsIstiod()
 	} else if features.PilotCertProvider == constants.CertProviderKubernetes {
-		log.Infof("initializing Istiod DNS certificates host: %s, custom host: %s", host, features.IstiodServiceCustomHost)
-		err = s.initDNSCerts()
+		// This will not work - better to fail Istiod startup so it can be detected early.
+		// Unlike other failures that we can recover from, this has no mitigation, user must
+		// choose a different source.
+		// The feature didn't work for few releases, but a skip-version upgrade may still
+		// encounter it.
+		log.Fatalf("PILOT_CERT_PROVIDER=kubernetes is no longer supported by upstream K8S", host, features.IstiodServiceCustomHost)
```

Comment:
why log `host, features.IstiodServiceCustomHost`

Reply:
Bad cut and paste, fixed. Also fixed the message when K8S RA is used to indicate the name and be different from istiod CA.